pkg/azure/azure_test.go

```diff
@@ -39,3 +43,90 @@ func Test_New(t *testing.T) {
 		})
 	}
 }
+
+func Test_RegisterCollectors(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockRegistry := mock_provider.NewMockRegistry(ctrl)
+
+	testCases := map[string]struct {
+		mockCollectors []*mock_provider.MockCollector
+		expectedErr    error
+	}{
+		"no collectors": {
+			mockCollectors: []*mock_provider.MockCollector{},
+		},
+		"AKS collector": {
+			mockCollectors: []*mock_provider.MockCollector{
+				mock_provider.NewMockCollector(ctrl),
+			},
+		},
+		"AKS and future storage collector": {
+			mockCollectors: []*mock_provider.MockCollector{
+				mock_provider.NewMockCollector(ctrl),
+				mock_provider.NewMockCollector(ctrl),
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			azProvider := &Azure{
+				logger:  testLogger,
+				context: parentCtx,
+			}
+			for _, c := range tc.mockCollectors {
+				call := c.EXPECT().Register(gomock.Any()).AnyTimes()
+				call.Return(nil)
+
+				azProvider.collectors = append(azProvider.collectors, c)
+			}
+
+			mockRegistry.EXPECT().MustRegister(gomock.Any()).Times(1)
+			err := azProvider.RegisterCollectors(mockRegistry)
+			assert.Equal(t, err, tc.expectedErr)
+		})
+	}
+}
+
+func Test_CollectMetrics(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	fakePromCh := make(chan prometheus.Metric)
+	testCases := map[string]struct {
+		mockCollectors []*mock_provider.MockCollector
+		expectedErr    error
+	}{
+		"base case": {
+			mockCollectors: []*mock_provider.MockCollector{
+				mock_provider.NewMockCollector(ctrl),
+			},
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			go func() {
+				for range fakePromCh {
+					// no metrics are generated here, so loop through to avoid
+					// process hang waiting on metrics that will never come
```

Comment:
Very small nit, put the comment before the for loop so we can get a nice tidy line:

`for range fakePromCh{}`